game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/CasualtyOrderOfLossesTestOnGlobal.java

```diff
@@ -367,9 +360,9 @@ void improvedArtillery() {
         CasualtyOrderOfLosses.sortUnitsForCasualtiesWithSupport(amphibAssault(attackingUnits));
```

Comment:
Drilling into this method, I see: `.defending(false)` , which implies this is offense. @trevan can you comment? Am I missing something?

Reply:
@DanVanAtta , in `UnitBattleComparator#compare`, it uses both the offense power and the defense power for comparing.  It will first compare the wanted direction (in this case `offense`) and if they are equal, it will then compare the other direction.  The marine has an offense of 2 (1 + marine bonus) and the artillery has an offense of 2 so they are equal.  It then compares the defense and before this change, the marine would have a defense of 3 (2 + marine bonus) and the artillery would only have a defense of 2.  The artillery would be considered worse and so would be the first in the list.  Now, they are considered identical (`compare` returns 0) so I *think* the behavior is that they get sorted according to how they previously exist in the list.  But it is possible the result could be non-deterministic.

I think maybe who ever wrote this test expects the artillery to support the marines but they only support infantry and mech_infantry.  So the marines get no support from them.

Reply:
Yeah, it looks like it is based on initial order of the list.  If I add the marines to `attackingUnits` before the artillery, then the resulting order is different.

Reply:
This ordering *should* be deterministic. Attack power, when on attack, should be the overriding attribute above any.

If we have a marine, on attack, with amphib, it has an attack power of 3. The artillery though has an attack power of 2. If choosing between marine (attack 3) and artillery (attack 2), the choice should *always* be artillery first. The ordering previously asserted in the test looks consistent with that.

Reply:
A marine on attack with amphib is only 2.  Why do you think it should be 3?

Reply:
Marine has a base of '2'.
The amphib assault bonus gives '+1', -> 3

Reply:
Does global have marines at a power of '1' on attack as base?

Reply:
Marines only have strength of 1 in global.

Reply:
Ah, yup, global has marines at '1', I was thinking big world v3 stats.